internal/v1/api.go

```diff
@@ -212,7 +212,8 @@ type ComposeResponse struct {
 
 // ComposeStatus defines model for ComposeStatus.
 type ComposeStatus struct {
-	ImageStatus ImageStatus `json:"image_status"`
+	ComposeRequest *ComposeRequest `json:"compose_request,omitempty"`
```

Comment:
after marking it as required, it should no longer be a pointer. `tools/prepare-source.sh` needs to be called again.

Reply:
Oh right I haven’t noticed that. We generate OpenAPI from Go. Rebased.

Reply:
Can we still rename it to `request` though, I really don't think it's a big change.
```
diff --git a/internal/v1/api.yaml b/internal/v1/api.yaml
index 7f6a812..0e16f2b 100644
--- a/internal/v1/api.yaml
+++ b/internal/v1/api.yaml
@@ -379,11 +379,11 @@ components:
     ComposeStatus:
       required:
         - image_status
-        - compose_request
+        - request
       properties:
         image_status:
           $ref: '#/components/schemas/ImageStatus'
-        compose_request:
+        request:
           $ref: "#/components/schemas/ComposeRequest"
     ImageStatus:
       required:
diff --git a/internal/v1/handler.go b/internal/v1/handler.go
index 5f09de1..7ca9d27 100644
--- a/internal/v1/handler.go
+++ b/internal/v1/handler.go
@@ -241,7 +241,7 @@ func (h *Handlers) GetComposeStatus(ctx echo.Context, composeId uuid.UUID) error
                        Status:       ImageStatusStatus(cloudStat.ImageStatus.Status),
                        UploadStatus: nil,
                },
-               ComposeRequest: &composeRequest,
+               Request: composeRequest,
        }
 
        if cloudStat.ImageStatus.UploadStatus != nil {

```

Reply:
Oh right it works, I thought I am unmarshaling it as a whole but actually this will work. Updated.